base_report_to_printer/report.py

```diff
@@ -40,6 +43,21 @@ def print_document(self, cr, uid, ids, report_name, html=None,
             )
         return printer.print_document(report, document, report.report_type)
 
+    def _can_print_report(self, cr, uid, ids, behaviour, printer, document,
+                          context=None):
+        """Predicate that decide if report can be sent to printer
+
+        If you want to prevent `get_pdf` to send report you can set
+        the `must_skip_send_to_printer` key to True in the context
+        """
+        if context is None:
+            context = self.pool['res.users'].context_get(cr, uid)
+        if context.get('must_skip_send_to_printer'):
+            return False
+        if behaviour['action'] == 'server' and printer and document:
+            return True
+        return False
+
```

Comment:
You're not using anything from the user's context, no need to get it.
The logic is confusing. How about changing the method body to a more direct form:
```
if context and context.get('must_skip_send_to_printer'):

    return behaviour['action'] == 'server' and printer and document
```